docs/docs/concepts/memory.md

```diff
@@ -132,44 +155,124 @@ def summarize_conversation(state: State):
+We dive into this more in [this section](#update-own-instructions)
 
-```python
-# Compile the graph with a checkpointer
-checkpointer = MemorySaver()
-graph = workflow.compile(checkpointer=checkpointer)
+**Remember a profile**
 
-# Invoke the graph with a thread ID
-config = {"configurable": {"thread_id": "1"}}
-graph.invoke(input_state, config)
+This technique is useful when there is specific information you may want to remember about a user/organization/group.
+You can define the schema of the profile ahead of time, and then use an LLM to update this based on interactions.
+We dive into this more in [this section](#remember-a-profile)
 
-# get the latest state snapshot at a later time
-config = {"configurable": {"thread_id": "1"}}
-graph.get_state(config)
-```
+**Remember a list**
+This technique is useful when you want repeatedly extract items and remember those.
+Similar to remembering a profile, you still define a custom schema to remember.
+The difference is that rather than remembering ONE schema per user/organization/group, you remember a list.
+We still use an LLM to update this list.
+We dive into this more in [this section](#remember-a-list)
+
+**Few shot examples**
+Sometimes you don't need to use an LLM to update memory, but rather can just raw interactions.
+You can then pull these raw interactions into the prompt as few-shot examples in future interactions.
+We dive into this more in [this section](#few-shot-examples)
+
+
+### How to update memory
+
+There are two main ways to update memory: "in the hot path" and "as a separate process".
+
+![](img/memory/hot_path_vs_process.png)
+
+#### Updating memory in the hot path
+
+This involves updating memory while the application is running. A concrete example of this is the way that ChatGPT does memory. ChatGPT can call tools to update or save a new memory, and it does that and then responds to the user. 
+
+This has several benefits. First of all, it happens realtime, so if the user starts a new thread right away that memory will be present. It also makes it possible to show the user that memory has been updated, and so can be a bit more transparent that way.
 
-Persistence is critical sustaining a long-running chat sessions. For example, a chat between a user and an AI assistant may have interruptions. Persistence ensures that a user can continue that particular chat session at any later point in time. However, what happens if a user initiates a new chat session with an assistant? This spawns a new thread, and the information from the previous session (thread) is not retained. This motivates the need for a memory service that can maintain data across chat sessions (threads). 
+This also has several downsides. It may slow down the final response since it needs to decide what to remember. It also means your application not only needs to think about the application logic, but also what to remember (which could result in more complicated instructions to the LLM).
```

Comment:
`but also what to remember (which could result in more complicated instructions to the LLM).`

Is this issue not at least partially shared by the application logic? In this case it's a bit less about what to remember, and more about what to recall or how to incorporate memories into the conversation. But both the learn and the recall path have to agree on the schema of memories